include/pmacc/algorithms/math/defines/pi.hpp

```diff
@@ -0,0 +1,47 @@
+/* Copyright 2018 Sergei Bastrakov
+ *
+ * This file is part of PMacc.
+ *
+ * PMacc is free software: you can redistribute it and/or modify
+ * it under the terms of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PMacc is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with PMacc.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#pragma once
+
+namespace pmacc
+{
+namespace algorithms
+{
+namespace math
+{
+
+    /** Values of pi and related constants as T_Type
+     */
+    template< typename T_Type >
+    struct Pi
+    {
+        static constexpr T_Type value = static_cast< T_Type >(
+            3.141592653589793238462643383279502884197169399
+        );
+        static constexpr T_Type halfValue = value / static_cast< T_Type >( 2.0 );
+        static constexpr T_Type quarterValue = value / static_cast< T_Type >( 4.0 );
+        static constexpr T_Type doubleReciprocalValue = static_cast< T_Type >( 2.0 ) / value;
```

Comment:
QUESTION:
Why do you expose a double reciprocal value but not a (single) reciprocal value?

Reply:
I assume from the code above you just implemneted the versions we used.
C provides various other constants. See https://www.gnu.org/software/libc/manual/html_node/Mathematical-Constants.html. 
But from my side, it is fine to implement them, when they are needed.

Reply:
Exactly, I was not sure if these are needed at all and so only added those that are used in PMacc.